build-support/bin/release.sh

```diff
@@ -94,194 +98,6 @@ function run_pex() {
-        'pants.backend.shell',\
-        'internal_plugins.releases',\
-      ]" \
-    "$@"
-}
-
-function build_3rdparty_packages() {
-  # Builds whls for 3rdparty dependencies of pants.
-  local version=$1
-
-  mkdir -p "${DEPLOY_3RDPARTY_WHEEL_DIR}/${version}"
-
-  local req_args=()
-  for req_file in "${REQUIREMENTS_3RDPARTY_FILES[@]}"; do
-    req_args=("${req_args[@]}" -r "${ROOT}/$req_file")
-  done
-
-  start_travis_section "3rdparty" "Building 3rdparty whls from ${REQUIREMENTS_3RDPARTY_FILES[*]}"
-  activate_tmp_venv
-
-  pip wheel --wheel-dir="${DEPLOY_3RDPARTY_WHEEL_DIR}/${version}" "${req_args[@]}"
-
-  deactivate
-  end_travis_section
-}
-
-function activate_tmp_venv() {
-  # Because the venv/bin/activate script's location is dynamic and not located in a fixed
-  # place, Shellcheck will not be able to find it so we tell Shellcheck to ignore the file.
-  # shellcheck source=/dev/null
-  VENV_DIR=$(mktemp -d -t pants.XXXXX) &&
-    "${PY}" -m venv "${VENV_DIR}" &&
-    "${VENV_DIR}/bin/pip" install wheel &&
-    source "${VENV_DIR}/bin/activate"
-}
-
-function pre_install() {
-  start_travis_section "SetupVenv" "Setting up virtualenv"
-  activate_tmp_venv
-  end_travis_section
-}
-
-function post_install() {
-  # this assume pre_install is called and a new temp venv activation has been done.
-  if [[ "${pause_after_venv_creation}" == "true" ]]; then
-    cat << EOM
-
-If you want to poke around with the new version of pants that has been built
-and installed in a temporary virtualenv, fire up another shell window and type:
-
-  source ${VENV_DIR}/bin/activate
-  cd ${ROOT}
-
-From there, you can run 'pants' (not './pants') to do some testing.
-
-When you're done testing, press enter to continue.
-EOM
-    read -r
-  fi
-  deactivate
-}
-
-function install_and_test_packages() {
-  local VERSION=$1
-  shift
-  local PIP_ARGS=(
-    "${VERSION}"
-    "$@"
-    --quiet
-    # Prefer remote or `--find-links` packages to cache contents.
-    --no-cache-dir
-  )
-
-  export PANTS_PYTHON_REPOS_REPOS="${DEPLOY_PANTS_WHEEL_DIR}/${VERSION}"
-
-  start_travis_section "wheel_check" "Validating ${VERSION} pantsbuild.pants wheels"
-  activate_twine
-  twine check "${PANTS_PYTHON_REPOS_REPOS}"/*.whl || die "Failed to validate wheels."
-  deactivate
-  end_travis_section
-
-  pre_install || die "Failed to setup virtualenv while testing ${NAME}-${VERSION}!"
-
-  # WONTFIX: fixing the array expansion is too difficult to be worth it. See https://github.com/koalaman/shellcheck/wiki/SC2207.
-  # shellcheck disable=SC2207
-  packages=(
-    $(run_packages_script list-packages | grep '.' | awk '{print $1}')
-  ) || die "Failed to list packages!"
-
-  for package in "${packages[@]}"; do
-    start_travis_section "${package}" "Installing and testing package ${package}-${VERSION}"
-    # shellcheck disable=SC2086
-    eval pkg_${package##*\.}_install_test "${PIP_ARGS[@]}" ||
-      die "Failed to install and test package ${package}-${VERSION}!"
```

Comment:
This line is when I realized how much I wanted to port from Bash to Python :) Glad to see it go